src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,103 @@
 package core.basesyntax;
 
-/**
- * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
- * реалізації ArrayList (default capacity, newCapacity...)</p>
- */
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 15;
+    private T[] storage;
+    private int size;
+
+    public ArrayList() {
+        this.storage = (T[]) new Object[DEFAULT_CAPACITY];
+        this.size = 0;
+    }
+
+    private void checkCapacity() {
+        if (storage.length == size) {
+            storage = ensureCapacity();
+        }
+    }
+
+    private T[] ensureCapacity() {
+        int newCapacity = storage.length + storage.length >> 1;
+        T[] newStorage = (T[]) new Object[newCapacity + 1];
```

Comment:
```suggestion
        T[] newStorage = (T[]) new Object[newCapacity];
```
or add 1 when calculating the value